www_src/pages/tinker/tinker.jsx

```diff
@@ -1,68 +1,157 @@
 var React = require('react');
-
 var render = require('../../lib/render.jsx');
+var api = require('../../lib/api');
+var elementTypes = require('../../components/el/el.jsx').types;
+var assign = require('react/lib/Object.assign');
 
 var Tabs = require('../../components/tabs/tabs.jsx');
 var ColorSpectrum = require('../../components/color-spectrum/color-spectrum.jsx');
-var Slider = require('../../components/range/range.jsx');
+var Loading = require('../../components/loading/loading.jsx');
+
+/**
+ * Tinker: Advanced editor for specific element attributes or styles
+ *
+ * Props
+ *   @key {boolean} isVisible Indicates if the Tinker Activity is currently visible (not paused)
+
+ * State
+ *   @key {boolean} loading If true, loading spinner is shown and UI is blocked
+ *   @key {object} element Represents flattened element data (via spec.flatten)
 
+ *   From router:
+ *   @key {object} params Contains params user, project, page, element, propertyName
+ */
 var Tinker = React.createClass({
+
+  mixins: [require('../../lib/router')],
+
   getInitialState: function () {
     return {
-      color: '#8FB70A'
+      loading: false,
+      element: null
     };
   },
-  onChangeColor: function (color) {
+
+  componentDidMount: function () {
+
+    // Prevent pull to refresh
+    document.body.style.overflowY = 'hidden';
```

Comment:
personally i think this would be better served by an API on the root component, but i'd rather not refactor in this ticket (I used the same strategy as is used elsewhere so it would be easily find-and-replaceable), mind if I file as a follow-up?